src/components/fields/date-field/date-field.tsx

```diff
@@ -48,36 +48,44 @@ export const DateField = (props: IGenericFieldProps<IDateFieldSchema>) => {
 		const minDateRule = validation?.find((rule) => "minDate" in rule);
 		const maxDateRule = validation?.find((rule) => "maxDate" in rule);
 		const excludedDatesRule = validation?.find((rule) => "excludedDates" in rule);
+		const withinDaysRule = validation?.find((rule) => "withinDays" in rule);
+		// determine the current date format to use
+		let currentDateFormat = dateFormat;
+		if (withinDaysRule?.["withinDays"].dateFormat) {
+			currentDateFormat = withinDaysRule?.["withinDays"].dateFormat;
+			setDateFormat(currentDateFormat);
+		}
```

Comment:
apologies, after some considerations, I think it is better to exclude `dateFormat` from the `withinDays` validation rule in date-field. my concern
- having the `dateFormat` in `IDateFieldValidationRule` override the one `IDateFieldSchema` is confusing, developers will not be aware of it
- the alternative of making developers ensure `dateFormat` in both `IDateFieldSchema` and `IDateFieldValidationRule` causes confusion because it is a hidden logic

so I propose to
- exclude `dateFormat` from the `withinDays` validation rule in date-field
- keep the `dateFormat` in conditional rendering can stay because it is applied on a different field, some synchronisation is expected

Reply:
so to use different dateFormat dev need to defined synchronisation in both place right ?